### ISSUE TITLE
redirect always got fallback=true

### ISSUE THREAD

**Issue by yih0620** [DAVIDhaker/django-sso#6]
I use 127.0.0.1:8000 as gateway and localhost:9000 as client.
I try to login gateway and redirect http://127.0.0.1:8000/__welcome/  success.
then i go to the address http://localhost:9000/sso/test/
it always redirect http://127.0.0.1:8000/__welcome/?fallback=true

the exception is in Service  f'{self.base_url}/sso/event/',
my base_url is 127.0.0.1:8000 but not client http://localhost:9000/
there's no url /sso/event/
is my setting wrong?

**Comment by DAVIDhaker** [comment]
Hi. You forget to attach gateway/client urls into your project(s). Please read Readme again. You need to add Django SSO urls to youproject/urls.py like it described in docs.

**Comment by DAVIDhaker** [comment]
Hi. May be 9000 port is wrong.

**Comment by DAVIDhaker** [comment]
Hi. You forget to attach gateway/client urls into your project(s). Please read Readme again. You need to add Django SSO urls to youproject/urls.py like it described in docs.

**Comment by yih0620** [comment]
yes. i'v already add urls to my projects like the described in docs. it's not work.

**Comment by DAVIDhaker** [comment]
Do you see anything in CLI? Library should report about errors.

**Comment by DAVIDhaker** [comment]
Please provide screenshot from CLI.

**Comment by yih0620** [comment]
gateway:
![image](https://user-images.githubusercontent.com/33826620/216261053-7cd4e1b5-00aa-42ec-8bfb-c69b93483350.png)
http://127.0.0.1:8000/login/  login admin OK.
![image](https://user-images.githubusercontent.com/33826620/216261064-6e4e2854-5591-475f-a736-fb895c346dda.png)

client:
![image](https://user-images.githubusercontent.com/33826620/216261208-fd0a6fd1-5320-4357-a21a-e289e9a3bd74.png)

goto http://localhost:9000/sso/test/
it always redirect http://127.0.0.1:8000/__welcome/?fallback=true
![image](https://user-images.githubusercontent.com/33826620/216261394-937c370d-2794-4858-b369-7b11294d7ed7.png)

sso/event error.
![image](https://user-images.githubusercontent.com/33826620/216261523-726ff88f-8227-4082-9f8e-547c347e90b8.png)

this is my client setting, and i am not defined the overrides.
![image](https://user-images.githubusercontent.com/33826620/216261699-2cca846c-0494-4b34-9424-6d28ffc3633d.png)

**Comment by DAVIDhaker** [comment]
No. Overrides are optional. On the client side you must provide SSO server link, token and inject urls to service-side project from `django_sso.sso_service.urls`.

**Comment by DAVIDhaker** [comment]
![image](https://user-images.githubusercontent.com/23550085/216266744-b03d95a1-0c65-4481-b7dc-a307e949b369.png)
SSO lib can't emit event to subordinated service because event aceptor url is not available - 404. Please show me your urls on service side.

**Comment by yih0620** [comment]
this is geteway urls.py:
![image](https://user-images.githubusercontent.com/33826620/216268775-57e86a2f-7505-4a25-86cd-dffca33e59aa.png)
this is client url.py:
![image](https://user-images.githubusercontent.com/33826620/216268789-1b075abe-c929-4c11-96f6-e45e0d2cae3a.png)

**Comment by DAVIDhaker** [comment]
- What is version of `Django SSO` are you use?
- What is version of `Django` are you used?
- Could you provide pure setup of Django's + the library (with same steps, what you do in your problematic project)?

For current i do not understand, how to occurs it.

**Comment by kakulukia** [comment]
@yih0620 you could check against https://github.com/kakulukia/test_client to see whats different in you soo client, but the 404 response clearly said the url on the client side is not reachable, maybe the urlpatterns you showed above is being overwritten afterwards?

You could test manually opening http://localhost:9000/sso/event/ in the browser - it should tell you:
Method Not Allowed: /sso/event/
[05/Feb/2023 12:37:03] "GET /sso/event/ HTTP/1.1" 405 0
since its just accepting POST requests

**Comment by yih0620** [comment]
> 
python version:3.7.4
django version: 3.2.17

I create new project today. 
https://github.com/yih0620/ssoServerTest
https://github.com/yih0620/ssoClientTest

![image](https://user-images.githubusercontent.com/33826620/216868585-f767753e-2c06-41a2-aa89-21e11eb62de7.png)

and thank you for your kind assistance!!

**Comment by kakulukia** [comment]
@DAVIDhaker shouldn't obtain and even be called on different hosts? It looks like even is called on the sso server instead of the client. Which log are we seeing @yih0620 ?

**Comment by yih0620** [comment]
@kakulukia log from ssoservertest.
i development on a single machine, and i set the hosts file as
![image](https://user-images.githubusercontent.com/33826620/216899784-20ba71c2-7f88-4abb-9716-978c24ddfece.png)

**Comment by yih0620** [comment]
i change SSO_ROOT = 'http://localhost:9000'
and reverse server and client ip
server at http://localhost:9000/
client at  127.0.0.1:8000
it work!
i think my Subordinated services ip set wrong before.

**Comment by DAVIDhaker** [comment]
@yih0620, hi! Excuse for long wait for reply. I think you solved problem on one's own. Little bit later i will try to test you project that you did published before.

**Comment by DAVIDhaker** [comment]
> @DAVIDhaker shouldn't obtain and even be called on different hosts? It looks like even is called on the sso server instead of the client. Which log are we seeing @yih0620 ?

There is no difference in direct communication between services. Only in browser required different hostnames. If i understand you question right.

**Comment by DAVIDhaker** [comment]
@yih0620 v2.0.0 is ready. You should update. And checkout new feature. :)

**Comment by yih0620** [comment]
Watching for file changes with StatReloader
Exception in thread django-main-thread:
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\thread
ing.py", line 926, in _bootstrap_inner
    self.run()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\thread
ing.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django\utils\autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django\core\management\commands\runserver.py", line 110, in inner_run
    autoreload.raise_last_exception()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django\utils\autoreload.py", line 87, in raise_last_exception
    raise _exception[1]
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django\core\management\__init__.py", line 375, in execute
    autoreload.check_errors(django.setup)()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django\utils\autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django\__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django\apps\registry.py", line 122, in populate
    app_config.ready()
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-p
ackages\django_sso\sso_gateway\apps.py", line 26, in ready
    affected_models = self.sso_settings.affected_models_fields.keys()
AttributeError: 'Settings' object has no attribute 'affected_models_fields'

update to 2.0 i have this error@@
and downgrade back to 1.1.4, it work well..

**Comment by kakulukia** [comment]
https://github.com/DAVIDhaker/django-sso/issues/8 << have a look at this to solve it

**Comment by CARocha** [comment]
sorry for put my problem here but i have this error

SSO token validation error
No connection adapters were found for '127.0.0.1:8000/sso/obtain/'
This error caused because SSO server don't know token of this service.

To solve this problem create subordinated service in SSO-server and put obtained token to SSO_TOKEN into settings.py file

in my settings file (sso_service_test)a put the token from my gateway_server like this
settings (sso_service_test1)
LOGIN_URL = '/login/'

SSO_ROOT = '127.0.0.1:8000'

SSO_TOKEN = 'kPeKnDUJg2uK4KGbCVzH4Cc342WHipLGWDp6wxooB1zFce37gpEdJWWtxxtUEshlTAruDfVHnNMnVf7rKjRBwBIgZRwOaV5S5v3D6cKObmlVrpCYBEqufu7cgYVH7e2AR'

in my server gateway add subordinate
<img width="1353" alt="Screen Shot 2023-02-19 at 5 41 35 PM" src="https://user-images.githubusercontent.com/168115/219982518-a1a47c61-31ed-4149-8e1e-80cb291759ab.png">

but no working this is versions on my gateway and service exactly (both)
asgiref==3.6.0
certifi==2022.12.7
charset-normalizer==3.0.1
Django==4.1.7
django-sso==2.0.0
idna==3.4
requests==2.28.2
sqlparse==0.4.3
urllib3==1.26.14

any idea?

**Comment by DAVIDhaker** [comment]
Hello @CARocha. For first. Try to install v3 (released minutes ago). If problem not solved - contact me via email for speed up problem solving.
